pkcs11/pkcs11_aescbc.c

```diff
@@ -87,16 +87,16 @@ int main(int argc, char* argv[])
     int slotId;
     int devId = 1;
 
-    if (argc != 5) {
+    if (argc != 4 && argc != 5) {
         fprintf(stderr,
-               "Usage: pkcs11_aescbc <libname> <slot> <tokenname> <userpin>\n");
+               "Usage: pkcs11_aescbc <libname> <slot> <tokenname> [userpin]\n");
         return 1;
     }
 
     library = argv[1];
     slot = argv[2];
     tokenName = argv[3];
-    userPin = argv[4];
+    userPin = (argc == 4) ? NULL : argv[4];
```

Comment:
Since you're using the length of this and want it to be 0 if it wasn't specified, can you assign it to an empty string here?  That way you don't need the extra conditional below.

Reply:
A PIN that is an empty string is different to a NULL pointer.
want to be sure that it is NULL and 0 to ensure an implementation of PKCS #11 doesn't complain.

Reply:
Makes sense, wasn't clear if the pin string itself was used anywhere.